src/dnsmasq_interface.c

```diff
@@ -212,26 +218,48 @@ void _FTL_new_query(const unsigned int flags, const char *name, const struct all
 	// Get domain pointer
 	domainsData* domain = getDomain(domainID, true);
 
-	// Try blocking regex if configured
-	if(domain->regexmatch == REGEX_UNKNOWN && blockingstatus != BLOCKING_DISABLED)
+	// Only check domains for blocking conditions when global blocking is enabled
+	char blockDomain = 0;
```

Comment:
This should be a boolean, including the function return type.

Reply:
The type `bool` isn't available in the majority of `dnsmasq`'s codebase. I didn't want to add `#include <stdbool.h>` all over the place for adding this type without any real benefit as the typical realization is a char on all major platforms IIRC.

Reply:
Would it be possible to use `bool` in our code, since it is just a macro?

Reply:
No, mixing is not possible. If we use `bool` in the `dnsmasq_interface.c` file, we will also need to use `bool` in the corresponding header file or the prototype will not match. However, we include the file `dnsmasq_interface.h` in `dnsmasq`'s code where `bool` is not available.
We guess should not `typedef char bool` on our own, I think this would be considered bad practice.

We *could* probably add
```c
#include <stdbool.h>
```
at the top of `dnsmasq_interface.h` and by this smuggle the header into `dnsmasq`'s code but I'd first have to test whether the code would still compile and also for possible unwanted side-effects (which I don't think would exist, however). If it turns out to work, we can use `bool` anywhere.

I'll investigate on this tomorrow.

Reply:
Done